docs/development/README.md

```diff
@@ -5,3 +5,53 @@ This is a developer docs directory. Not meant for end users.
 ## Contents
 
 - [Update component configuration.](./updating-components.md)
+
+## Spell checking
+
+As part of CI process, we run [codespell](https://github.com/codespell-project/codespell) to find potential
+typos in our code base. This section describes how to configure and use and it.
+
+### Running locally
+
+To run `codespell` locally, install it either using the [official method](https://github.com/codespell-project/codespell#installation)
+or use package manager specific to your Linux distribution.
+
+With `codespell` installed, run the following command to run spell checking locally:
+```sh
+make codespell
+```
+
+It should find potential typos in the code.
+
+### Ignoring specific words
+
+Sometimes, `codespell` will detect some words as typos, like `AKS`, `ACI` or `IAM`, which will cause the CI to
+report errors. To resolve that, we define our list of ignored words in `.codespell.ignorewords` file,
+which is then passed to `codespell` via `--ignore-words` flag.
+
+If you find that some word you added should be ignored, add it to this file.
+
+### Ignoring vendored files
+
+Lokomotive ships some 3rd party code like vendored Go dependencies and Helm charts, which are maintained
+by other people and may also contain some typos. It is recommended to configure `codespell` to **SKIP** those
+files from checking, instead of **FIXING** them in our repository, to avoid deriving from upstream code, which
+may make the update process of this code more difficult in the future (merge conflicts, extensive diffs etc.).
+
+To skip some file or directory (recursively) from spell checking, add the path to `.codespell.skip` file.
```

Comment:
```suggestion
To recursively skip some file or directory from spell checking, add the path to `.codespell.skip` file.
```

Reply:
It implies "recursively skipping files", which does not make much sense to me :smile:

Reply:
Fair enough, I'll resolve.